contracts/allocators/OnsenAllocatorV2.sol

```diff
@@ -0,0 +1,174 @@
+        // reads
+        uint256 index = tokenIds[id];
+        IERC20 LPtoken = _tokens[index];
+        uint256 balance = LPtoken.balanceOf(address(this));
+
+        // interactions
+        //Deposit LP token into onsen, if deposit succesfull this address should have in return sushi tokens
+        if (balance > 0) {
+            IERC20(LPtoken).approve(masterChef, balance);
+            IMasterChef(masterChef).deposit(id, balance); // deposit into Onsen
+        }
+
+        // Stake the sushi tokens
+        if (IERC20(sushi).balanceOf(address(this)) > 0) {
+            enterSushiBar(true); // manage rewards
+        }
+
+        //Calculate gains/loss
+        // Retrieve current balance for pool and address
+        UserInfo memory currentUserInfo = IMasterChef(masterChef).userInfo(id, address(this));
+
+        uint256 last = extender.getAllocatorAllocated(id) + extender.getAllocatorPerformance(id).gain;
+
+        if (currentUserInfo.amount >= last) {
+            gain = uint128(currentUserInfo.amount - last);
+        } else {
+            loss = uint128(last - currentUserInfo.amount);
+        }
+    }
+
+    /**
+        Asuming the index of the amounts [] can be use as parameter for the mapping tokenIds
+        Example:
+        amounts => [20,50,150]
+        So the values that will be used to retrieve the id used in onsen will be 0, 1 , 2
+    */
+    function deallocate(uint256[] memory amounts) public override onlyGuardian {
+        for (uint256 i; i < amounts.length; i++) {
+            uint256 amount = amounts[i];
+            if (amount > 0) {
+                IMasterChef(masterChef).withdraw(tokenIds[i], amount); // withdraw from Onsen
+            }
+        }
+    }
+
+    function _deactivate(bool panic) internal override {}
+
+    function _prepareMigration() internal override {}
+
+    function amountAllocated(uint256 id) public view override returns (uint256) {
+        UserInfo memory currentUserInfo = IMasterChef(masterChef).userInfo(id, address(this));
+        return currentUserInfo.amount;
+    }
+
+    function rewardTokens() public view override returns (IERC20[] memory) {
+        IERC20[] memory rewards = new IERC20[](1);
+        rewards[0] = IERC20(sushi);
+        return rewards;
+    }
+
+    function utilityTokens() public view override returns (IERC20[] memory) {
+        IERC20[] memory utility = new IERC20[](2);
+        utility[0] = IERC20(sushi);
+        utility[1] = IERC20(xSushi);
+        return utility;
+    }
+
+    function name() external pure override returns (string memory) {
+        return "OnsenAllocator";
+    }
+
+    /* ========== INTERNAL FUNCTIONS ========== */
+
+    /**
+     * @notice stake sushi rewards if enter is true else return funds to treasury.
+     * @param _stake bool
+     */
+    function enterSushiBar(bool _stake) internal {
+        uint256 balance = IERC20(sushi).balanceOf(address(this));
+        if (balance > 0) {
+            if (!_stake) {
```

Comment:
Would check for true case first. It's odd to check for false case here, considering false is the less common case.

Reply:
This function was updated to just stake any sushi tokens available, so check is no longer needed.